Middlewares/verifyUserReq.js

```diff
@@ -0,0 +1,50 @@
+const UserModel = require("../Models/user.model");
+
+exports.verifyUserRequest = async(req, res, next) => {
+    if (!req.body.name) {
+        return res.status(400).send({
+            message: "Name is Required"
+        })
+    }
+
+    if (!req.body.userId) {
+        return res.status(400).send({
+            message: "UserId is Required"
+        })
+    }
+
+    const user = await UserModel.findOne({ userId: req.params.userId })
+    if (user) {
+        return res.status(400).send({
+            message: 'UserId already used!'
+        })
+    }
+
+    if (!req.body.email) {
+        return res.status(400).send({
+            message: "EmailId is Required"
+        })
+    }
+
+
+    const email = await UserModel.findOne({ emailId: req.body.emailId })
```

Comment:
const name should not be email, it will confuse later. Better to call it userByEmail.